Task12 whack a mole

EstebanMadrigalAK opened the pull request:
# Pull Request
## Description
This branch include a new application called Wack a Mole on the folder "Task12-whack-a-mole" but refactored, using sass and implementing the next new features:
- Add one more row of moles
- Add kick.wav sound on successful and hihat.wav on failed attempt 
- Add a scorecard 
- Add an input to enter the player's name
- Save the best 5 scores
- Add a level
- Add an image to the cursor 
- Show the time to the player

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?
I test the functionality of the application with help of an extension called live server that helps to see the changes of the html and how it looks.

1. install the extension
2. open the html file you want to test on vsc
3. select the go live option on the bottom panel
4. it will open the web browser showing the html
5. every change on the html or css is going to be showed on the browser window

EstebanMadrigalAK committed:
Adding initial files

EstebanMadrigalAK committed:
Scss implemented

EstebanMadrigalAK committed:
Adding One row more

EstebanMadrigalAK committed:
Sound added

EstebanMadrigalAK committed:
Adding constant values

EstebanMadrigalAK committed:
removing duplicated elements

EstebanMadrigalAK committed:
Adding scorecard, level and cursor icon

EstebanMadrigalAK committed:
timer added

EstebanMadrigalAK committed:
Style refactored

EstebanMadrigalAK committed:
code formatted

falvarado3 commented on Task12-whack-a-mole/index.html:
```diff
@@ -0,0 +1,72 @@
+    </div>
+  </div>
+  <div class="game">
+    <div class="hole hole1">
+      <div class="mole"></div>
+    </div>
+    <div class="hole hole2">
+      <div class="mole"></div>
+    </div>
+    <div class="hole hole3">
+      <div class="mole"></div>
+    </div>
+    <div class="hole hole4">
+      <div class="mole"></div>
+    </div>
+    <div class="hole hole5">
+      <div class="mole"></div>
+    </div>
+    <div class="hole hole6">
+      <div class="mole"></div>
+    </div>
+    <div class="hole hole7">
+      <div class="mole"></div>
+    </div>
+    <div class="hole hole8">
+      <div class="mole"></div>
+    </div>
+    <div class="hole hole9">
+      <div class="mole"></div>
+    </div>
```
I would recommend moving this repeated code to JS if you want.
You can select the parent element of those divs and inject a generated HTML code to it, you can use template literals, so you can create a loop to create the 9 holes into a string:

```
[...Array(9)].forEach((_, index) => {
  htmlToBeInserted += `
    <div class="hole hole${index}">
      <div class="mole"></div>. 
    </div>`; 
});  
parentElement.insertAdjacentHTML('afterend',  htmlToBeInserted);
```

falvarado3 commented on Task12-whack-a-mole/src/game.js:
```diff
@@ -0,0 +1,131 @@
+                timeText.textContent = timeText.textContent - 1;
+                this.updateTime();
+            }
+        }, 1000)
```
missing semicolon

falvarado3 commented on Task12-whack-a-mole/src/game.js:
```diff
@@ -0,0 +1,131 @@
+        timeUp = false;
+        this.peep();
+        this.updateTime();
+        setTimeout(() => this.finishGame(), gameTime)
```
missing semicolon

falvarado3 commented on Task12-whack-a-mole/src/game.js:
```diff
@@ -0,0 +1,131 @@
+    }
+
+    this.sortPlayers = () => {
+        players.sort((player1, player2) => {
+            if (player1.score < player2.score) {
+                return 1;
+            }
+            else if (player1.score > player2.score) {
+                return -1;
+            }
+            return 0;
+        });
```
When you sort an array and the compare function involves numbers, you can simply do:
`players.sort((player1, player2) => player1.score - player2.score) `
It would will you the same result

falvarado3 commented on Task12-whack-a-mole/src/game.js:
```diff
@@ -0,0 +1,131 @@
+
+    this.bonk = e => {
+        if (!e.isTrusted) return;
+        let audioToPlay = failAudio
```
missing semicolon

EstebanMadrigalAK committed:
Changer for PR review feedback

Add a function to insert repeated code
Sort function improved
Add semicolon where are missing

EstebanMadrigalAK committed:
Merge branch 'main' into Task12-Whack-a-Mole
